timely/src/dataflow/operators/core/capture/replay.rs

```diff
@@ -118,8 +117,9 @@ where
                     activator.activate_after(delay);
                 }
 
-                output.cease();
-                output.inner().produced().borrow_mut().drain_into(&mut progress.produceds[0]);
+                use timely_communication::Push;
+                output.push(&mut None);
```

Comment:
I think we should either push `&mut None` and remove `Push::done`, or use `Push::done` consistently. No big deal, but removing a function might not be a bad call.

Reply:
Similarly, `Push` has a `send` function that we rarely use. Consider removing it.

Reply:
A related thing: there were moments where one needs to know `C: Container` earlier than you might want, on account of we need a `Push<_>` implementation to be able to call `done()`, even though `C` is not involved. Potentially a `Done` trait that provides the method, and then `Push: Done` or something. Maybe overly complicated, but removes at least one generic parameter from a type that doesn't seem to need it. Will investigate!

Reply:
I wouldn't mind a `Done` trait to avoid spreading the types from drop to the type and everywhere from there. I also think that we should keep the stack of pushers for exactly the reason that we might _not_ want to force a tee on everyone.

Reply:
It's been a while: https://github.com/TimelyDataflow/timely-dataflow/pull/508